File: EuroMillionsRandom.py
import random

# For EuroMillions five main numbers, between 1 and 50, and two Lucky Stars, between 1 and 12. The function takes 2 inputs mainNumbers,luckyStars

def generateNums(mainNumbers,luckyStars):
    newNum = []
    luckyNum = []
    for i in range(mainNumbers):
        newNum.append(random.randint(1,50))
    for i in range(luckyStars):
        num = random.randint(1,12)
        if num not in luckyNum:
            luckyNum.append(num)
        else:
            num = random.randint(1,12)
            luckyNum.append(num)
        
    return newNum+luckyNum


# print(generateNums(5,2))

File: EuroMillionsRandom.js
// create random number function

function createRandomNum( n ) {
    let random = Math.random();
    random = Math.floor( random * n ) + 1;
    return random
}

function generateNums (firstNumbersCount,secondNumbersCount){
    let newNum = [];
    let luckyNum = [];

    for (let i = 0 ; i < 5 ; i ++){
        newNum.push( createRandomNum (firstNumbersCount) );
    }

    for (let i = 0 ; i < 2 ; i ++){
        firstPick = createRandomNum (secondNumbersCount);
        if (luckyNum.includes( firstPick )){
            firstPick = createRandomNum (secondNumbersCount);
            luckyNum.push( firstPick );
        } else {
            luckyNum.push( firstPick );
        }
    }

  // combine two lists
    return newNum.concat( luckyNum ) ;
}

const euroMillionsMain = generateNums ( 50 , 12 ).slice ( 0 , 5 );
const euroMillionsLucky = generateNums ( 50 , 12 ).slice ( 5 , 8 );

let euroMillions = "First number: "+(euroMillionsMain.toString()) + ' ' + "\nLucky number: " + euroMillionsLucky.toString() 

console.log(euroMillions)
